src/main/java/gov/cabinetoffice/gap/applybackend/service/SubmissionService.java

```diff
@@ -215,6 +222,17 @@ public void submit(final Submission submission, final GrantApplicant grantApplic
         }
 
         submission.setGapId(generateGapId(grantApplicant.getId()));
+        try {
```

Comment:
This could be its own method. If you highlight all the code down to the bottom of the `try...catch` block, `right click in intellij` -> `refactor` -> `extract method` then the IDE will create the method with all required params for you and you just need to give it a name.

After that, I would wrap the method call in a check to see if the `submission.getApplication().getScheme().getVersion() > 1` so that we don't need to have an exception handled for any outstanding V1 submissions.